x-pack/test/api_integration/deployment_agnostic/apis/observability/ai_assistant/complete/functions/get_dataset_info.spec.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MessageAddEvent, MessageRole } from '@kbn/observability-ai-assistant-plugin/common';
+import expect from '@kbn/expect';
+import { LogsSynthtraceEsClient } from '@kbn/apm-synthtrace';
+import { last } from 'lodash';
+import { GET_RELEVANT_FIELD_NAMES_SYSTEM_MESSAGE } from '@kbn/observability-ai-assistant-plugin/server/functions/get_dataset_info/get_relevant_field_names';
+import { ChatCompletionStreamParams } from 'openai/lib/ChatCompletionStream';
+import {
+  LlmProxy,
+  RelevantField,
+  createLlmProxy,
+} from '../../../../../../../observability_ai_assistant_api_integration/common/create_llm_proxy';
+import { getMessageAddedEvents, getSystemMessage, systemMessageSorted } from './helpers';
+import type { DeploymentAgnosticFtrProviderContext } from '../../../../../ftr_provider_context';
+import { createSimpleSyntheticLogs } from '../../synthtrace_scenarios/simple_logs';
+
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const log = getService('log');
+  const observabilityAIAssistantAPIClient = getService('observabilityAIAssistantApi');
+  const synthtrace = getService('synthtrace');
+
+  describe('get_dataset_info', function () {
+    this.tags(['failsOnMKI']);
+    let llmProxy: LlmProxy;
+    let connectorId: string;
+
+    before(async () => {
+      llmProxy = await createLlmProxy(log);
+      connectorId = await observabilityAIAssistantAPIClient.createProxyActionConnector({
+        port: llmProxy.getPort(),
+      });
+    });
+
+    after(async () => {
+      llmProxy.close();
+      await observabilityAIAssistantAPIClient.deleteActionConnector({
+        actionId: connectorId,
+      });
+    });
+
+    // Calling `get_dataset_info` via the chat/complete endpoint
+    describe('POST /internal/observability_ai_assistant/chat/complete', function () {
+      let messageAddedEvents: MessageAddEvent[];
+      let logsSynthtraceEsClient: LogsSynthtraceEsClient;
+      let getRelevantFields: () => Promise<RelevantField[]>;
+
+      const USER_MESSAGE = 'Do I have any Apache logs?';
+
+      before(async () => {
+        logsSynthtraceEsClient = synthtrace.createLogsSynthtraceEsClient();
+        await createSimpleSyntheticLogs({ logsSynthtraceEsClient });
+
+        void llmProxy.interceptWithFunctionRequest({
+          name: 'get_dataset_info',
+          arguments: () => JSON.stringify({ index: 'logs*' }),
+          when: () => true,
+        });
+
+        ({ getRelevantFields } = llmProxy.interceptSelectRelevantFieldsToolChoice());
+
+        void llmProxy.interceptConversation(`Yes, you do have logs. Congratulations! 🎈️🎈️🎈️`);
```

Comment:
why are we not awaiting this?

Reply:
ah  never mind, we just need the await on `waitForAllInterceptorsToHaveBeenCalled`

Reply:
The interceptors are setup before making the call to `POST /internal/observability_ai_assistant/chat/complete`. Awaiting here would never return (it would timeout)